17-dimensionality_reduction.Rmd

```diff
@@ -10,132 +11,567 @@
+      
+    ggforce::geom_autodensity(alpha = .3) +
+    ggforce::facet_matrix(vars(-class), layer.diag = 2) + 
+      
     scale_color_brewer(palette = "Dark2") + 
     scale_fill_brewer(palette = "Dark2")
 }
 ```
 
-- PCA = unsupervised method, finds up to N new features (where N = # features) to explain variation.
 
-```{r 17-pca}
-bean_rec_trained %>%
+### PCA - Principal Component Analysis
+
+This is the first technique, and it is the starting point for the others as it is unsupervised.
+PCA considers the data as in a matrix made of homogeneous elements, and it computes the calculation for making groups with regards of their variance level within each variable. 
+
+
+- linear
+- unsupervised method, finds up to N new features (where N = # features) to explain variation
+- account for variance
+- it is a method for writing a matrix X of rank r as a sum of r matrices of rank 1: X= M1 + M2 + ... + Mr
+- hypothesises correlation among predictors
+
+
+What is aji ? 
+
+PCj = (aj1 ×Predictor 1)+(aj2 ×Predictor 2)+···+(ajP ×Predictor P)
+
+
+```{r, cache=TRUE}
+pca_val <- bean_rec_trained %>%
   step_pca(all_numeric_predictors(), num_comp = 4) %>%
-  plot_validation_results() + 
+  plot_validation_results() + # the function
   ggtitle("Principal Component Analysis")
+
+pca_val
 ```
 
-## Partial Least Squares (PLS)
+The next plot involves a comparison of the components. The book uses a function for plotting PCA bars of the major components, with the help of the {learntidymodels} package.
+
+      learntidymodels::plot_top_loadings()
+
+There is another way to turn around the use of this package. I will do it the way Julia Silge did in the [video: Dimensionality reduction with tidymodels for the Billboard Top 100 ](https://www.youtube.com/watch?v=kE7H1oQ2rY4) and make a comparison with PLS further below.
+
+
+```{r, cache=TRUE}
+pca <- bean_rec_trained %>%
+  step_pca(all_numeric_predictors(), num_comp = 4) %>% 
+  prep() %>% 
+  tidy(number=4) %>%
+  filter(component%in% paste0("PC",1:4)) %>%
+  group_by(component) %>%
+  slice_max(abs(value),n=5) %>%
+  ungroup() %>%
+  ggplot(aes(abs(value),terms,fill=value>0))+
+  geom_col(alpha=0.8)+
+  facet_wrap(vars(component),scales="free_y") +
+  labs(fill="PCA Positive?",
+       x="",
+       y="")+
+  theme_bw()+
+  theme(legend.position = "top")
+```
 
-- Supervised PCA.
 
-```{r 17-pls}
-bean_rec_trained %>%
-  step_pls(all_numeric_predictors(), outcome = "class", num_comp = 4) %>%
+### PLS - Partial Least Squares
+
+
+It is a technique which is very similar to PCA.
+
+PCA is a part of PLS, in the meaning that in addition to PCA, PLS makes use of the **outcome**. It practically makes the components considering the outcome values.
+
+
+
+```{r echo=FALSE, fig.align='center', fig.cap='[credits: PARTIAL LEAST-SQUARES REGRESSION: A TUTORIAL](http://home.mit.bme.hu/~horvath/IDA/1-s2.0-0003267086800289-main.pdf)', out.width='100%'}
+knitr::include_graphics(here::here('images/pls.png'))
```

Comment:
```suggestion
knitr::include_graphics("images/pls.png")
```
In this context, it looks like `here::here` just confuses things. The relative path should do it. I'm going to try to accept these changes if I can so we can hopefully make this work.